### DISCUSSION TITLE
Hosting brand images

### DISCUSSION THREAD

**Discussion by bryceco** [osmlab/name-suggestion-index#4792] [General]
Having the imageURL is great, but for me relying on a URL has a few drawbacks:
* Some images are in SVG/BMP format, which iOS doesn't support, so I can't display them
* I'm sometimes downloading images that are much larger than necessary for what I'm presenting to the user
* I download the NSI at build time, so it's possible for the URL to change to change and become unaccessible
* Downloading the images at build-time and including them with the app unnecessarily bloats the size of the app (2x-5x depending on what image resolution I pick).

Questions:
* Are there copyright issues with storing the images on a server to download at runtime? 
* Is it against GitHub's TOS to commit the images with my project, and fetch them from GitHub?
* Is there any interest in NSI hosting the images itself?

**Comment by bhousel** [comment]
> * Are there copyright issues with storing the images on a server to download at runtime?
> * Is it against GitHub's TOS to commit the images with my project, and fetch them from GitHub?
> * Is there any interest in NSI hosting the images itself?

These are great questions!  My understanding on this today is that 
- The images are definitely copyrighted by their owners, which is why I'd prefer for applications to fetch them at runtime.  This leaves the actual control over the images with the owners, and as long as an application uses the images in a [fair-use](https://en.wikipedia.org/wiki/Fair_use) way (e.g. identification), that application's use of them should be fine.
- I don't believe including assets or linking to them is against the [GitHub TOS](https://docs.github.com/en/free-pro-team@latest/github/site-policy), but if we were to mirror them in GitHub (or anywhere), copyright owners are allowed to submit DMCA takedown notices to get their content removed - see [GitHub's policy on this](https://docs.github.com/en/free-pro-team@latest/github/site-policy/dmca-takedown-policy).  The bar for submitting a DMCA takedown is _very_ low - people submit them all the time.
- Nobody is going to be better at running a global image CDN than giants like Facebook, Twitter, and Wikimedia
- Fetching the images from third party sites like Facebook and Twitter has raised privacy concerns on the iD repo and elsewhere, but I don't believe these hosts actually receive the kinds of information about users that some people in the community have suggested.  There is a lot of misinformation out there.

So I don't think that having them in _this_ repo is something I want to do... 
But I or someone could definitely create a dedicated repo to preserve the images (but with the understanding that DMCA could shut it down).  At one time, OSMF had also floated the idea of hosting the images on openstreetmap.org, but I'm pretty sure they didn't think this through wrt DMCA, file sizes, refresh frequency, or anything, they just wanted to make the inflated community privacy concerns go away.

Worth mentioning is that GitHub has some guidelines on [how big a repo can be](https://github.community/t/working-with-large-files-and-repositories/10203) and [how to check your file size](https://docs.github.com/en/free-pro-team@latest/github/managing-large-files/what-is-my-disk-quota) - currently I believe they discourage repos above 1GB, and things start breaking if you test those limits.

We have close to 7000 logo-d items in the NSI right now, so as long as we are smart about file sizes, it would be under the limit for a while, but it's something to think about.

Related issues in the iD repo (toxic OSM community trigger warning):
https://github.com/openstreetmap/iD/issues/7040
https://github.com/openstreetmap/iD/issues/7028
https://github.com/openstreetmap/iD/issues/6361
https://github.com/openstreetmap/iD/issues/5167

**Comment by bryceco** [comment]
Thanks for the detailed response. I've decided to host the images on my own server.

**Comment by 1ec5** [comment]
> Are there copyright issues with storing the images on a server to download at runtime?

@bryceco, in case it helps for your purposes, each of the images coming from Wikimedia Commons is freely licensed. In fact, most of them are in the public domain, subject only to trademark restrictions, due to simplicity. The Facebook and Twitter images are a mixed bag: some may also be too simple to be eligible for copyright, but there’s no automated way to filter down to just those images. (That’s one more argument in favor of the way NSI is currently doing things, but you may find the public domain images more comfortable to work with.)

> Some images are in SVG/BMP format, which iOS doesn't support, so I can't display them

The Facebook and Twitter images are in JPEG format, while the Commons images are a mix of SVG, PNG, and occasionally JPEG. If you encounter an SVG image from Commons, Commons has [an API method](https://commons.wikimedia.org/wiki/Special:ApiSandbox#action=query&format=json&prop=pageimages&titles=File%3AMcDonald%27s%20Golden%20Arches.svg%7CFile%3ABurger%20King%20logo.svg%7CFile%3ARally%27s%20logo.svg&piprop=thumbnail&pithumbsize=50) for getting PNG thumbnails of a batch of images at a given size. I’d trust Commons’s SVG rasterizer over any other rasterizer because Commons contributors generally ensure that the rasterized SVG previews look good on the site.
